File: crates/g8-extractor/tests/fixtures/python/has_annotations.py
# PASS-detect fixture: Python file with G8 magic-comment annotations.
# Accepts both # prefix (Python-native) and // prefix (uniform cross-language).

# @g8.capability(name = "http-handler", status = "in_flight", substrate = "net")
def handle_request(request):
    pass


# @g8.capability(name = "auth-validator", status = "landed", substrate = "auth")
def validate_token(token: str) -> dict:
    return {}


# @g8.convergence_test(for_capability = "http-handler", scenario = "post-request")
def test_post_request():
    pass


# @g8.intent(description = "HTTP request handling", substrate = "net")
_intent_marker = None


# @g8.plan(title = "rate-limiting", status = "Idea", substrate = "net")
def _plan_marker():
    pass

File: crates/g8-extractor/tests/fixtures/rust/has_annotations.rs
// PASS-detect fixture: Rust file with G8 magic-comment annotations.

// @g8.capability(name = "http-fetch", status = "in_flight", substrate = "net")
pub fn fetch_url(url: &str) -> Vec<u8> {
    vec![]
}

// @g8.capability(name = "json-parse", status = "landed", substrate = "data")
pub fn parse_json(bytes: &[u8]) -> serde_json::Value {
    serde_json::Value::Null
}

// @g8.convergence_test(for_capability = "http-fetch", scenario = "timeout")
#[test]
fn test_http_fetch_timeout() {}

// @g8.intent(description = "HTTP networking utilities", substrate = "net")
mod net {}

// @g8.plan(title = "streaming-export", status = "Idea", substrate = "net")
fn placeholder_streaming_export() {}

// @g8.decision(title = "Use reqwest for HTTP", status = "accepted")
fn placeholder_decision() {}
